frontend/src/app/core/service/dark-mode.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+/* the service provides a centralized and reusable way to manage dark mode functionality throughout the app. */
+export class DarkModeTogglerService {
+  private darkModeSubject: BehaviorSubject<boolean>;
+  public darkMode$: BehaviorSubject<boolean>;
```

Comment:
this should rather be an observable

Reply:
I'm curious about the viability of using Observable in scenarios involving dialogs and popups, where it's crucial to know the current state of dark mode to apply the dark theme dynamically.  Wouldn't BehaviorSubject offer more flexibility for such scenarios?  Perhaps you know something that I don't..